nixos/modules/services/networking/netmaker/netmaker.nix

```diff
@@ -0,0 +1,625 @@
+          Access-Control-Max-Age ${builtins.toString cfg.cors.maxAge}
+          -Server
+        }
+        reverse_proxy http://${cfg.api.internal.addr}
+      '';
+    })
+    (lib.mkIf (cfg.webserver.type == "caddy" && cfg.ui.enable) {
+      # see https://github.com/gravitl/netmaker/blob/630c95c48b43ac8b0cdff1c3de13339c8b322889/docker/Caddyfile#L1-L20
+      services.caddy.virtualHosts."https://${cfg.ui.domain}".extraConfig = ''
+        ${lib.optionalString (cfg.email != "") "tls ${cfg.email}"}
+        header {
+          Access-Control-Allow-Origin https://${cfg.ui.domain}
+          Access-Control-Max-Age ${builtins.toString cfg.cors.maxAge}
+          Strict-Transport-Security "max-age=31536000;"
+          X-XSS-Protection "1; mode=block"
+          X-Frame-Options "SAMEORIGIN"
+          X-Robots-Tag "none"
+          -Server
+        }
+
+        handle /nmui-config.js {
+          header Content-Type text/javascript
+          respond <<EOF
+          window.NMUI_AMUI_URL='${cfg.ui.saas.amuiUrl}';
+          window.NMUI_INTERCOM_APP_ID='${cfg.ui.saas.intercomAppID}';
+          window.NMUI_BACKEND_URL='https://${cfg.api.domain}';
+          EOF
+        }
+
+        handle {
+          # see https://github.com/gravitl/netmaker-ui-2/blob/b305a96fc19160346f7cd59439d4625f262aaedf/nginx.conf#L7-L7
+          try_files {path} {path}/ /index.html
+          root * ${cfg.ui.package}/var/www
+          file_server
+        }
+      '';
+    })
+    (lib.mkIf (cfg.db.type == "sqlite") { })
+    (lib.mkIf (cfg.mqtt.type == "mosquitto" && cfg.webserver.type == "caddy") {
+      # see https://www.reddit.com/r/selfhosted/comments/14wqwa4/comment/jrp07gq/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button
+      services.caddy.virtualHosts."https://${cfg.mqtt.domain}".extraConfig = ''
+        ${lib.optionalString (cfg.email != "") "tls ${cfg.email}"}
+        reverse_proxy http://${cfg.mqtt.internal.addr} {
+          stream_timeout 6h
+          stream_close_delay 1m
+        }
+      '';
+    })
+    (lib.mkIf (cfg.mqtt.type == "mosquitto") {
+      systemd.services.mosquitto.after = [ "netmaker-configure.service" ];
+      systemd.services.mosquitto.requires = [ "netmaker-configure.service" ];
+      systemd.services.netmaker.after = [ "mosquitto.service" ];
+      systemd.services.netmaker.requires = [ "mosquitto.service" ];
+
+      services.mosquitto.enable = true;
+      services.mosquitto.listeners = [{
+        address = cfg.mqtt.internal.host;
+        port = cfg.mqtt.internal.port;
+        users."${cfg.mqtt.username}" = {
+          passwordFile = cfg.mqtt.passwordFile;
+          acl =
+            if !cfg.mqtt.acl.restricted then [
+              "readwrite #"
+            ] else [
+              # search for `client.Subscribe` in:
+              # - https://github.com/gravitl/netmaker
+              # - https://github.com/gravitl/netclient
+              # https://github.com/gravitl/netclient
```

Comment:
Why are these commented out ?

Reply:
those were supposed to be instructions on finding out set of topics used by Netmaker